manifests/kustomize/env/aws/iam/aws-configurations-patch.yaml

```diff
@@ -0,0 +1,78 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: workflow-controller-configmap
+  namespace: kubeflow
+data:
+  config: |
+    {
+    executorImage: $(executorImage),
+    artifactRepository:
+    {
+        archiveLogs: true,
+        s3: {
+            bucket: $(artifactRepositoryBucket),
+            keyPrefix: $(artifactRepositoryKeyPrefix),
+            endpoint: s3.amazonaws.com,
+            insecure: false
+        }
+    }
+    }
+---
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: ml-pipeline-ui
+spec:
+  template:
+    spec:
+      containers:
+        - name: ml-pipeline-ui
+          env:
+            - name: ARGO_ARCHIVE_LOGS
+              value: "true"
+            - name: ARGO_ARCHIVE_ARTIFACTORY
+              value: s3
+            - name: ARGO_ARCHIVE_BUCKETNAME
+              value: $(artifactRepositoryBucket)
+            - name: ARGO_ARCHIVE_PREFIX
+              value: $(artifactRepositoryKeyPrefix)
+            - name: VIEWER_TENSORBOARD_POD_TEMPLATE_SPEC_PATH
+              value: /etc/config/viewer-tensorboard-template.json
+          volumeMounts:
+            - name: config-volume
+              mountPath: /etc/config
+      volumes:
+        - name: config-volume
+          configMap:
+            name: ml-pipeline-ui-viewer-template
+---
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: ml-pipeline
+spec:
+  template:
+    metadata:
+      annotations:
+        iam.amazonaws.com/role: $(awsIAMRole)
```

Comment:
Let's make the `key` configurable?  `iam.amazonaws.com/role` is for k2iam users. As EKS release feature https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/, we have lots of users use `eks.amazonaws.com/role-arn` as well.

Reply:
kustomize don't allow templating for keys yet - i.e. I cannot do `$(key): $(value)` for annotation.

a possible workaround is to use `commonAnnotations` (actually this is what I used for my deployment, cuz I was lazy. I gave all kfp deployments the same iam role)